src/internal/passkeysBrowser.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * This file contains the underlying implementations for exposed API surface in
+ * the {@link api/passkeysBrowser}. By moving the methods out into a separate file,
+ * other namespaces and processes can access these methods without depending on the entire
+ * faucet namespace and without having a dependency cycle error.
+ */
+import { sha3_256 } from "@noble/hashes/sha3";
+import { p256 } from "@noble/curves/p256";
+import { AptosConfig } from "../api/aptosConfig";
+import { Hex } from "../core";
+import {
+  AnyRawTransaction,
+  deriveTransactionType,
+  getAuthenticatorForWebAuthn,
+  getSigningMessage,
+} from "../transactions";
+import { HexInput, PendingTransactionResponse } from "../types";
+import { submitTransaction } from "./transactionSubmission";
+
+export async function signWithPasskey(args: {
+  aptosConfig: AptosConfig;
+  credentialId: HexInput;
+  publicKey: HexInput;
```

Comment:
Would you already have the passkey / other information, or do you need to pull this out of elsewhere?